Makefile

```diff
@@ -262,7 +262,9 @@ lint-container-test-deps: ## Check that the test-container module only imports a
 	@cd test/integration/consul-container && \
 		$(CURDIR)/build-support/scripts/check-allowed-imports.sh \
 			github.com/hashicorp/consul \
-			internal/catalog/catalogtest
+			internal/catalog/catalogtest \
+			internal/catalog \
+			internal/resource
```

Comment:
IIUC the need for these packages is only so that we can pass the created type registry into the  `resourcetest.Resource` method. If so then I think I would prefer we take a different approach. When I created the `resourcetest.resourceBuilder` I had it set some default tenancy because the resource service would error out if I didn’t. I would rather the builder leave it unspecified with the service filling it in using its own type registry to help determine what the correct defaults are. That should eliminate the need for most of the changes in this PR and for needing to thread a registry everywhere. Besides making it easier to do testing without having to manage resource type registries, its also just a good UX feature to have. Users could avoid needing to specify tenancies when defaults should apply. For now thats probably defaulting things to the right scope but in the not too distant future we should also do ACL based tenancy defaulting.

I think it would also be a good idea for the resource service to zero out the tenancy fields which shouldn’t apply to the resource. That way if someone using the API is always setting tenancy fields (via query params etc.) then they don’t get errors that would force them to special case their requests. We ran into issues with this and the current tenancy specification with the k8s integration a while back. It caused bugs and subsequent work in our k8s controllers to special case sometimes setting namespaces (enterprise) and sometimes not (community)

Reply:
Checking the codebase, I see there are zero calls to `resourceBuilder.WithTenancy(..)`. To strike a compromise I'm thinking something that supports both variations for now and leaves the preferred behavior intact.

```
// Original signature remains intact. New callers should start using `WithTenancy(...)`
func Resource(rtype *pbresource.Type, name string) *resourceBuilder {
	return DefaultResource(rtype, name, nil)
}

// Existing tests in this PR can easily be search/replaced to use this method. Open to suggestions for a better name.
//
// DefaultResource creates a resource with tenancy defaulted to the type's scope.
func DefaultResource(rtype *pbresource.Type, name string, registry resource.Registry) *resourceBuilder {
	tenancy := nil
	// Use registry to create default tenancy for passed in type.
	if registry != nil {
		reg, ok := registry.Resolve(rtype)
		if !ok {
			panic(fmt.Sprintf("You forgot to register type %v in the registry", rtype))
		} 
		tenancy = resource.DefaultTenancyFor(reg.Scope),
	}

	return &resourceBuilder{
		resource: &pbresource.Resource{
			Id: &pbresource.ID{
				Type: &pbresource.Type{
					Group:        rtype.Group,
					GroupVersion: rtype.GroupVersion,
					Kind:         rtype.Kind,
				},
				Tenancy: tenancy,
				Name:    name,
			},
		},
	}
}
```

The UX defaults you mention are already in place on the resource service endpoints. Since these are tests, I would prefer test writers be explicit with tenancy.

Reply:
IMO unless the test author is using `WithTenancy` then there is nothing explicit about tenancy. Whether your test relies on magic defaults from the registry or magic defaults from the resource service, it isn't materially different from the p.o.v. of test authoring. Therefore, I don't think there is much here to warrant the extra dependencies and complexity introduced by needing to create a registry with all the right types and to thread it through to every helper method that may need to interact with the resource service.

My suggestion would be to back out the changes to all usage of `resourcetest.Resource` to remove the extra registry parameter and instead change that method to not specify tenancy.

Reply:
We'll have to allow nil tenancy across the board then which is currently required from the original implementation from a few months ago. I'll refactor that into the resource service and then revisit this PR.